ci/build/build-docker-images.py

```diff
@@ -881,5 +885,60 @@ def main(
         # push_readmes(build_type is MERGE)
 
 
+def fix_docker_images(
+    image: str = "ray-ml",
+    version: str = "nightly",
+    repo: str = DOCKER_HUB_REPO,
+):
+    tags = create_image_tags(
+        image_name=image,
+        py_versions=list(PY_MATRIX.keys()),
+        image_types=list(BASE_IMAGES.keys()),
+        specific_tag=None,  # Set to `latest` for latest image fixes
+        version=version,
+        suffix=None,
+    )
+    print(dict(tags))
+
+    # Pull images we want to rebuild
+    for base_tag in tags:
+        base_image = f"{repo}/{image}:{base_tag}"
+
+        print(f"docker pull {base_image}")
+
+    # Re-tag these base images as e.g. pinned/ray-ml:tag
+    # This is so we can re-run the build command safely.
+    pinned_base_image = {}
+    for base_tag in tags:
+        base_image = f"{repo}/{image}:{base_tag}"
+        pinned_image = f"pinned/{image}:{base_tag}"
+
+        pinned_base_image[base_image] = pinned_image
+
+        print(f"docker tag {base_image} {pinned_image}")
```

Comment:
do we actually need to tag here?

Reply:
See comment:

```
    # Re-tag these base images as e.g. pinned/ray-ml:tag
    # This is so we can re-run the build command safely.
```

If we don't tag here, we can only run the script once. If this fails or our fix is insufficient, the next build would build on the already updated docker image - adding more layers if more iterations are used.

Think of this as a temporary variable. It's also just metadata, so very cheap